IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -584,7 +601,32 @@ - (BOOL)clearCacheForAccount:(MSIDAccountIdentifier *)accountIdentifier
         query.environmentAliases = aliases;
         query.matchAnyCredentialType = YES;
 
-        BOOL result = [_accountCredentialCache removeCredetialsWithQuery:query context:context error:error];
+        NSError *credentialRemovalError;
+        BOOL result = [_accountCredentialCache removeCredetialsWithQuery:query context:context error:&credentialRemovalError];
+        
+        if (!result)
+        {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, context, @"Failed to remove credentials with error %@", MSID_PII_LOG_MASKABLE(credentialRemovalError));
+            if (error) *error = credentialRemovalError;
+        }
+        
+        if (clearAccounts)
+        {
+            MSIDDefaultAccountCacheQuery *accountQuery = [MSIDDefaultAccountCacheQuery new];
+            accountQuery.homeAccountId = homeAccountId;
+            accountQuery.environmentAliases = aliases;
+            accountQuery.accountType = MSIDAccountTypeMSSTS;
+            
+            NSError *accountRemovalError;
+            result = [_accountCredentialCache removeAccountsWithQuery:accountQuery context:context error:&accountRemovalError];
+            
+            if (!result)
+            {
+                MSID_LOG_WITH_CTX_PII(MSIDLogLevelWarning, context, @"Failed to remove accounts with error %@", MSID_PII_LOG_MASKABLE(accountRemovalError));
+                if (error) *error = accountRemovalError;
+            }
+        }
+        
```

Comment:
Am not able to comment on the exact line for this one but would it be better to return the result variable instead from this method.

Reply:
Addressed.